### DISCUSSION TITLE
Sampling question for multivariate outcomes

### DISCUSSION THREAD

**Discussion by novawz** [kogalur/randomForestSRC#227] [Q&A]
I have a question regarding the randomForestSRC package and I’m hoping you can provide me any insight on it. I’m trying to use the multivariate RF to deal with multiple outcomes. For example, if I have a very large number of outcomes, e.g. 1000 variables contained in the outcome, it is not very efficient to throw all the outcome variables in the multivariate rf model. So I’m wondering if it would be more efficient to select a smaller subset of the outcomes for each tree when fitting the model. Since I tried 100-200 outcome variables in the multivariate RF and it is much faster.

Then I tried paralleling 200 times on the rf model with only 1 tree fitted and sampled a smaller subset of the outcome variables in each tree and average the results to get my model. And I did get a very similar result as I used all outcomes to fit one rf model with 200 trees, and it is more efficient than using the original multivariate rf as I expected. However, I noticed from your Vignettes that it is not recommended to use parallel outside the rfsrc model. So is there a way that I can try this directly from the rfsrc package?

Thank you so much for your time reading my question and I look forward to your reply!

**Comment by ishwaran** [comment]
Thanks for your interesting question.  What you need is random outcome selection similar to random feature selection that is used.  Therefore we need to create a `mtry` parameter for multivariate forests.  Currently there is actually a `ytry` parameter that functions like this but it only applies for unsupervised forests.  We will see if we can extend `ytry` to the supervised multivariate forest setting and maybe have that ready for the next build.
